src/main/java/com/google/step/youtube/DescriptionParser.java

```diff
@@ -52,14 +54,34 @@ public Pattern getPattern() {
      * @param description of the YouTube video to be parsed.
      * @return A list of all matches found in description.
      */
-    public static List<String> findMatches(Pattern pattern, String description) {
-        List<String> matches = new ArrayList<>();
+    public static List<OfferSnippet> findMatches(Pattern pattern, String description) {
+        List<OfferSnippet> matches = new ArrayList<>();
         Matcher matcher = pattern.matcher(description);
 
         while (matcher.find()) {
-            matches.add(matcher.group());
+            matches.add(OfferSnippet.create(matcher.group(), getSnippet(matcher.start(), description)));
         }
         return matches;
     }
 
+    // Finds the snippet (line) of description which conatins the promocode index. 
+    // Bounds the snippet at MAX_SNIPPET_LENGTH characters and does not truncate words.
+    private static String getSnippet(int promocodeIndex, String description) {
```

Comment:
for classes, methods use `/* */`. (see the java style guide.)

Reply:
Changed.